### ISSUE TITLE
Cannot start on Android 7

### ISSUE THREAD

**Issue by tux-mind** [cSploit/android#831]
### DEVICE:

Nexus 5X

### OS version:

7.1.1 ( API 25 )

### cSploit version:

1.7.0-unstable+ce3185f

### busybox (installed or not):

not again, just rooted

### Rooted with supersu?

yep

### logcat (filter it!):

```
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]: java.lang.NullPointerException: Attempt to get length of null array
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.net.Endpoint.<init>(Endpoint.java:62)
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.net.Target.setEndpoint(Target.java:626)
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.net.Target.<init>(Target.java:479)
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.core.System.reset(System.java:973)
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.core.System.uncaughtReloadNetworkMapping(System.java:38
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.core.System.init(System.java:199)
02-11 11:03:08.989 10498 10498 E CSPLOIT[core.System.errorLogging]:     at org.csploit.android.CSploitApplication.onCreate(CSploitApplication.java:72)
```

### Related Issues:
Probably a lot.

### Cause:

`java.net.NetworkInterface#getHardwareAddress` returns `null`, w/o exceptions, it seems a security restriction related error ( `NetPermission("getNetworkInformation")` required )

refs:

[NetworkInterface.java](https://android.googlesource.com/platform/libcore/+/nougat-mr1.5-release/ojluni/src/main/java/java/net/NetworkInterface.java#433)
[NetworkInterface.c](https://android.googlesource.com/platform/libcore/+/nougat-mr1.5-release/ojluni/src/main/native/NetworkInterface.c#166)

### Notes:

fix with love <3

**Comment by pahapoika** [comment]
Strange.
It works perfectly on Nougat (Oxygenos)
On my op3t.

**Comment by IwraStudios** [comment]
Made a branch for the fix: #832
